hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -193,21 +194,26 @@ protected static function getContractDates($contracts) {
     // If there are two (or more) Contracts starting on the same day
     // then we pick only the one with latest End date.
     foreach ($contracts['values'] as $contract) {
-      $details = civicrm_api3('HRJobDetails', 'getsingle', array(
+      $details = civicrm_api3('HRJobDetails', 'get', array(
```

Comment:
no need to write anything .. any BAO class extends a DAO class which also extends CRM_Core_DAO which contain some handy methods such as **commonRetrive** which can be used to get the data , something like this :

https://github.com/civicrm/civihr/blob/1c5b932a7aef97690fd6cc95c7d4f974e3a3ad3d/com.civicrm.hrjobroles/tests/phpunit/HrJobRolesTestBase.php#L123